### ISSUE TITLE
Upgrade from C122 -> C123 on `Win x64` via `1.65.x - Nightly` (Checking C123)

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#36637]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [x] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using
```
Brave | 1.65.62 Chromium: 123.0.6312.29 (Official Build) nightly (64-bit)
-- | --
Revision | 9adf65558b12ae895bddf7568f3b7247d6da39f4
OS | Windows 11 Version 23H2 (Build 22631.3155)
```

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x86` using the following build(s):

```
Brave | 1.65.63 Chromium: 123.0.6312.29 (Official Build) nightly (32-bit)
-- | --
Revision | 0388fcfccedfa84b7be8f813a7a15bff4066f0b2
OS | Windows 10 Version 22H2 (Build 19045.4123)
```

<details>
<summary>C122 --> C123 on-demand upgrade check - PASSED</summary>

* installed `1.65.60 Chromium: 122.0.6261.111` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.65.63 Chromium: 123.0.6312.29` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|---------------
![image](https://github.com/brave/brave-browser/assets/2602313/4e277b05-3f7c-4c4c-987b-95cf13919b22) | ![image](https://github.com/brave/brave-browser/assets/2602313/3ab8680c-dd10-450a-9dda-8fa37773613a) | ![image](https://github.com/brave/brave-browser/assets/2602313/7b8bc304-f3e5-4cd8-857e-eb118c9c0112) | ![image](https://github.com/brave/brave-browser/assets/2602313/bea81724-5fcc-44aa-b9ac-b83ecb0d3250)

</details>

<details>
<summary>C123 --> C123 on-demand upgrade check - PASSED</summary>

* installed `1.65.62 Chromium: 123.0.6312.29` via `BraveBrowserStandaloneNightlySetup32_123_1_65_62.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.64.57 Chromium: 122.0.6261.29` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/989241db-661a-4091-bc8e-99638031c01f) | ![image](https://github.com/brave/brave-browser/assets/2602313/afb76bde-1da2-43ef-971b-7966a3320326) | ![image](https://github.com/brave/brave-browser/assets/2602313/a76a33a7-08ac-4c14-81e3-46170acce1ce) | ![image](https://github.com/brave/brave-browser/assets/2602313/90fb0255-d3bf-4727-9dcd-9092d6e0582d)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[03/08/24 16:26:31.948][BraveUpdate:goopdate][7772:5832][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{65EF548E-1BD2-4C47-95F7-0BAE40C1864D}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 10]
[03/08/24 16:26:31.948][BraveUpdate:goopdate][7772:5832][Running installer][C:\Program Files\BraveSoftware\Update\Install\{65EF548E-1BD2-4C47-95F7-0BAE40C1864D}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[03/08/24 16:26:49.201][BraveUpdate:goopdate][7772:5832][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[03/08/24 16:26:49.201][BraveUpdate:goopdate][7772:5832][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 30][][]
[03/08/24 16:26:49.209][BraveUpdate:goopdate][7772:5832][Started process][9780]
[03/08/24 16:26:49.213][BraveUpdate:goopdate][7772:5832][LogEventHelper][4][12][1][[
Application update/install.
pid=7772, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=123.1.65.62, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=123.1.65.62, Install result=0x00000000
]][]
[03/08/24 16:26:49.218][BraveCrashHandler][9780:9604][ConfigManager::LoadGroupPolicies][Machine is not Enterprise Managed]
[03/08/24 16:26:49.219][BraveCrashHandler][9780:9604][ConfigManager::LoadGroupPolicies][No Group Policies found under key][HKLM\Software\Policies\BraveSoftware\Update\]
[03/08/24 16:26:49.219][BraveCrashHandler][9780:9604][OmahaPolicyManager::set_policy][Group Policy][[CachedOmahaPolicy][is_initialized][0][is_managed][0][auto_update_check_period_minutes][-1][download_preference][][cache_size_limit][-1][cache_life_limit][-1][updates_suppressed][-1][-1][-1][proxy_mode][][proxy_server][][proxy_pac_url][][install_default][-1][update_default][-1]]
[03/08/24 16:26:49.222][BraveCrashHandler][9780:9604][CrashHandler::RunUntilShutdown]
```

</details>

<details>
<summary>C122 --> C123 background upgrade check - PASSED</summary>

* installed `1.65.60 Chromium: 122.0.6261.111` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{8382614B-C053-4467-A3A4-16291E0223CF}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.65.63 Chromium: 123.0.6312.29`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background`

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|---------------|-----------------
![image](https://github.com/brave/brave-browser/assets/2602313/62d09906-4b7b-4c32-a086-5cf00576ce49) | ![image](https://github.com/brave/brave-browser/assets/2602313/801eac07-b770-44fa-95e4-d4908b8cabf5) | ![image](https://github.com/brave/brave-browser/assets/2602313/6a31f9d5-4e84-4ea1-ae94-55642a13c90d) | ![image](https://github.com/brave/brave-browser/assets/2602313/37771e73-2544-4246-aec2-153e968985cb)

</details>

<details>
<summary>C123 --> C123 background upgrade check - PASSED</summary>

* installed `1.64.54 Chromium: 122.0.6261.29` via `BraveBrowserStandaloneNightlySetup32_122_1_64_54.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{0CB2B8C6-95E1-42D7-99AE-FA9E276ECCE9}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.65.63 Chromium: 123.0.6312.29`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/cdfcdb0e-7865-4c6a-ae35-6c2a24b8806a) | ![image](https://github.com/brave/brave-browser/assets/2602313/29257cad-407f-47dd-b387-834058e4a163) | ![image](https://github.com/brave/brave-browser/assets/2602313/a958b4c9-7cfa-4d90-8c0a-6f784d35a504) | ![image](https://github.com/brave/brave-browser/assets/2602313/fdc95b62-a5af-471f-b409-2e5b7555a29b)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[03/08/24 16:42:53.583][BraveUpdate:goopdate][8748:6296][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{3FDD0FE7-C4FF-494C-B495-00CFE4A9259C}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 0]
[03/08/24 16:42:53.583][BraveUpdate:goopdate][8748:6296][Running installer][C:\Program Files\BraveSoftware\Update\Install\{3FDD0FE7-C4FF-494C-B495-00CFE4A9259C}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[03/08/24 16:43:14.496][BraveUpdate:goopdate][8748:6296][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[03/08/24 16:43:14.496][BraveUpdate:goopdate][8748:6296][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 2][]["C:\Program Files\BraveSoftware\Brave-Browser-Nightly\Application\brave.exe" --from-installer]
[03/08/24 16:43:14.496][BraveUpdate:goopdate][8748:6296][Started process][8736]
[03/08/24 16:43:14.496][BraveUpdate:goopdate][8748:6296][LogEventHelper][4][12][1][[
Application update/install.
pid=8748, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=123.1.65.62, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=123.1.65.62, Install result=0x00000000
]][]
[03/08/24 16:43:14.511][BraveCrashHandler][8736:4828][ConfigManager::LoadGroupPolicies][Machine is not Enterprise Managed]
[03/08/24 16:43:14.511][BraveCrashHandler][8736:4828][ConfigManager::LoadGroupPolicies][No Group Policies found under key][HKLM\Software\Policies\BraveSoftware\Update\]
[03/08/24 16:43:14.511][BraveCrashHandler][8736:4828][OmahaPolicyManager::set_policy][Group Policy][[CachedOmahaPolicy][is_initialized][0][is_managed][0][auto_update_check_period_minutes][-1][download_preference][][cache_size_limit][-1][cache_life_limit][-1][updates_suppressed][-1][-1][-1][proxy_mode][][proxy_server][][proxy_pac_url][][install_default][-1][update_default][-1]]
[03/08/24 16:43:14.511][BraveCrashHandler][8736:4828][CrashHandler][Instance is already running][32-bit][1]
[03/08/24 16:43:14.590][BraveUpdate:goopdate][8748:5780][Started process][11808]
[03/08/24 16:43:14.590][BraveUpdate:goopdate][6728:7632][DllEntry exit][0x00000000]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][DllEntry]["C:\Program Files\BraveSoftware\Update\BraveUpdate.exe" /ping PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48cmVxdWVzdCBwcm90b2NvbD0iMy4wIiB1cGRhdGVyPSJPbWFoYSIgdXBkYXRlcnZlcnNpb249IjEuMy4zNjEuMTQ5IiBzaGVsbF92ZXJzaW9uPSIxLjMuMzYxLjE0OSIgaXNtYWNoaW5lPSIxIiBzZXNzaW9uaWQ9IntCMjI3NUU3Qy02OTVFLTRGMDgtQTZEMy01RjczMUQyQkY3MzZ9IiB1c2VyaWQ9IntCNDVDOUY2Qy1DMjE5LTQ5QjAtOUQ3OS0wRjE5NjZGRDcyQ0R9IiBpbnN0YWxsc291cmNlPSJzY2hlZHVsZXIiIHRlc3Rzb3VyY2U9ImF1dG8iIHJlcXVlc3RpZD0iezc4RDI5MDg5LUJGREMtNDg5NC1CRTc3LUU4NDhDMzE4QTIzRn0iIGRlZHVwPSJjciIgZG9tYWluam9pbmVkPSIwIj48aHcgcGh5c21lbW9yeT0iMiIgc3NlPSIxIiBzc2UyPSIxIiBzc2UzPSIxIiBzc3NlMz0iMSIgc3NlNDE9IjEiIHNzZTQyPSIxIiBhdng9IjEiLz48b3MgcGxhdGZvcm09IndpbiIgdmVyc2lvbj0iMTAuMC4xOTA0NS40MTIzIiBzcD0iIiBhcmNoPSJ4ODYiLz48YXBwIGFwcGlkPSJ7QzZDQjk4MUUtREIzMC00ODc2LTg2MzktMTA5Rjg5MzM1ODJDfSIgdmVyc2lvbj0iMTIzLjEuNjUuNjIiIG5leHR2ZXJzaW9uPSIxMjMuMS42NS42MyIgYXA9Im5pZ2h0bHkiIGxhbmc9IiIgYnJhbmQ9IkdHTFMiIGNsaWVudD0iIiBpbnN0YWxsYWdlPSIwIj48ZXZlbnQgZXZlbnR0eXBlPSIxMiIgZXZlbnRyZXN1bHQ9IjEiIGVycm9yY29kZT0iMCIgZXh0cmFjb2RlMT0iMCIvPjxldmVudCBldmVudHR5cGU9IjEzIiBldmVudHJlc3VsdD0iMSIgZXJyb3Jjb2RlPSIwIiBleHRyYWNvZGUxPSIwIi8-PGV2ZW50IGV2ZW50dHlwZT0iMTQiIGV2ZW50cmVzdWx0PSIxIiBlcnJvcmNvZGU9IjAiIGV4dHJhY29kZTE9IjAiIGRvd25sb2FkZXI9ImJpdHMiIHVybD0iaHR0cHM6Ly91cGRhdGVzLWNkbi5icmF2ZS5jb20vZGVsdGEvQnJhdmUtQnJvd3Nlci1OaWdodGx5L25pZ2h0bHkvd2luLzEyMy4xLjY1LjYzLzEyMy4xLjY1LjYyL3g4Ni9icmF2ZV9pbnN0YWxsZXItZGVsdGEtaWEzMi5leGUiIGRvd25sb2FkZWQ9IjM1NjIwMDgiIHRvdGFsPSIzNTYyMDA4IiBkb3dubG9hZF90aW1lX21zPSIxNDM2MiIvPjxldmVudCBldmVudHR5cGU9IjE0IiBldmVudHJlc3VsdD0iMSIgZXJyb3Jjb2RlPSIwIiBleHRyYWNvZGUxPSIwIi8-PGV2ZW50IGV2ZW50dHlwZT0iMTUiIGV2ZW50cmVzdWx0PSIxIiBlcnJvcmNvZGU9IjAiIGV4dHJhY29kZTE9IjAiLz48ZXZlbnQgZXZlbnR0eXBlPSIzIiBldmVudHJlc3VsdD0iMSIgZXJyb3Jjb2RlPSIwIiBleHRyYWNvZGUxPSIxOTY3MDciIHNvdXJjZV91cmxfaW5kZXg9IjAiIHVwZGF0ZV9jaGVja190aW1lX21zPSI1NTciIGRvd25sb2FkX3RpbWVfbXM9IjE0NDQxIiBkb3dubG9hZGVkPSIzNTYyMDA4IiB0b3RhbD0iMzU2MjAwOCIgaW5zdGFsbF90aW1lX21zPSIyMDkxMyIvPjwvYXBwPjwvcmVxdWVzdD4]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][ConfigManager::LoadGroupPolicies][Machine is not Enterprise Managed]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][ConfigManager::LoadGroupPolicies][No Group Policies found under key][HKLM\Software\Policies\BraveSoftware\Update\]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][OmahaPolicyManager::set_policy][Group Policy][[CachedOmahaPolicy][is_initialized][0][is_managed][0][auto_update_check_period_minutes][-1][download_preference][][cache_size_limit][-1][cache_life_limit][-1][updates_suppressed][-1][-1][-1][proxy_mode][][proxy_server][][proxy_pac_url][][install_default][-1][update_default][-1]]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][C:\Program Files\BraveSoftware\Update\1.3.361.149\goopdate.dll][version 1.3.361.149][opt][dev]
[03/08/24 16:43:14.605][BraveUpdate:goopdate][11808:11668][is machine: 1]
[03/08/24 16:43:14.980][BraveUpdate:goopdate][11808:11668][DllEntry exit][0x00000000]
[03/08/24 16:43:14.980][BraveUpdate:goopdate][8748:7116][DllEntry exit][0x00000000]
```

</details>
